### DISCUSSION TITLE
Improvements to internal page linking

### DISCUSSION THREAD

**Discussion by rraineygc** [outline/outline#2788] [Ideas]
There are 2 improvements that I have found would greatly improve the experience when linking internal pages

## Auto Searching for Matching Text
If I highlight a piece of text, I am currently prompted to create a page with that text's name or manually type to search for an existing page.

If we were to be shown a list of pages that matched that highlighted text, it would make it a lot more intuitive to link internal pages

## Display page parent on search results when adding an internal link
Further to the first item above, when displaying a list of matching pages (have searched manually or had them popup based on a matching string), it would vastly improve the experience to show the page's parent in the results to differentiate between pages across collections with the same name.

Now:
![image](https://user-images.githubusercontent.com/7805625/142611971-7ef55d6a-47b0-47a4-a7fe-24b70a26b14a.png)

Example:
![image](https://user-images.githubusercontent.com/7805625/142612123-92032946-7cda-493c-827c-89a5705e495e.png)

**Comment by MajinV** [comment]
This would be great to help adoption for new editors across the wider team and resolve one of the (very few) clunky ux pieces people struggle with

**Reply by tommoor** [reply]
Agreed, we'll be moving the editor codebase into the main Outline codebase this week – it's currently separate – this will allow for tighter integrations such as this
